fastestimator/trace/adapt/reduce_lr_on_plateau.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import inspect
+from typing import Optional, Union
+
+import numpy as np
+import tensorflow as tf
+import torch
+
+from fastestimator.backend import get_lr, set_lr
+from fastestimator.summary import System
+from fastestimator.trace.trace import Trace
+from fastestimator.util.data import Data
+
+
+class ReduceLROnPlateau(Trace):
+    """Reduce learning rate based on evaluation results
+
+    Args:
+        model: model instance
+        monitor: The monitor metric name, if None, model's validation loss will be used as metric.
+        patience: Number of epochs to wait before reducing LR again.
+        factor: Reduce factor of learning rate. Defaults to 0.1.
+        reduce_mode: higher is better (max) or lower is better (min). Defaults to min
+        min_lr: minimum learning rate, defaults to 1e-6.
+    """
+    system: System
+
+    def __init__(self,
+                 model: Union[tf.keras.Model, torch.nn.Module],
+                 metric: Optional[str] = None,
+                 patience: int = 10,
+                 factor: float = 0.1,
+                 reduce_mode: str = "min",
```

Comment:
'reduce_mode' name isn't super intuitive - at first it thought it might be controlling whether learning rate increases or decreases

Reply:
done